client/state/notices/actions.js

```diff
@@ -36,10 +36,11 @@ export function createNotice( status, text, options = {} ) {
 	};
 }
 
-export function setRoute( path ) {
+export function setRoute( path, query = {} ) {
```

Comment:
In #6306 (to be merged shortly), this action creator is moved to `state/ui/actions.js`